FirebasePerformance/Sources/Gauges/FPRGaugeManager.m

```diff
@@ -117,7 +117,9 @@ - (BOOL)gaugeCollectionEnabled {
 }
 
 - (void)startCollectingGauges:(FPRGauges)gauges forSessionId:(NSString *)sessionId {
-  [self prepareAndDispatchGaugeData];
+  dispatch_async(self.gaugeDataProtectionQueue, ^{
+    [self prepareAndDispatchGaugeData];
+  });
 
   self.currentSessionId = sessionId;
```

Comment:
Will old gauge data be associated with the new session ID if we set it now before batching? Is it fine if so?

Reply:
Very good question and a great catch. I have modified the code to handle that scenario better. Please take a look.